### PR TITLE
TCVP-977 added scheduled task in oraface-api to populate redis

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

[TCVP-977](https://justice.gov.bc.ca/jira/browse/TCVP-977)

This PR adds a Scheduled Task to the oracle-data-interface project to automatically refresh the Statute table in Redis.  Eventually this data will come from JUSTIN, but for now it's just pulled from a static CSV file.

- added a Scheduled Task with a cron job to trigger a codetable refresh. 
- added a **/codetable/refresh** endpoint to manually trigger a codetable refresh.
- added a refreshAtStartup option to trigger a codetable refresh on application start.
- added JAVA_OPTS to the Dockerfile to easily pass in JVM arguments if needed (like logging)
- updated README.md, added new ENV variables to adjust local and deployment environments
-- REDIS_HOST
-- REDIS_PORT
-- REDIS_PASSWORD
-- CODETABLE_REFRESH_ENABLED
-- CODETABLE_REFRESH_CRON
-- JAVA_OPTS

See src\backend\README.md for ENV variables descriptions.

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

This command brings up the project and autopopulates the Redis instance.
`docker-compose up -d --build oraface-api`
```
20:36:44.705 [main]  INFO  ca.bc.gov.open.jag.tco.oracledatainterface.OracleDataInterfaceApplication: Started OracleDataInterfaceApplication in 8.069 seconds (JVM running for 8.841)
20:36:44.710 [main]  DEBUG ca.bc.gov.open.jag.tco.oracledatainterface.scheduled.CodeTableTasks: Refreshing code tables at startup.
20:36:44.711 [main]  DEBUG ca.bc.gov.open.jag.tco.oracledatainterface.service.LookupService: Refreshing code tables in redis.
20:36:51.878 [main]  INFO  ca.bc.gov.open.jag.tco.oracledatainterface.OracleDataInterfaceApplication: Application up. Log level set to 'DEBUG'
20:37:00.000 [scheduling-1] DEBUG ca.bc.gov.open.jag.tco.oracledatainterface.scheduled.CodeTableTasks: Scheduled 'codeTableRefresh' cron job called.
20:37:00.002 [scheduling-1] DEBUG ca.bc.gov.open.jag.tco.oracledatainterface.service.LookupService: Refreshing code tables in redis.
```

## Does the change impact or break the Docker build?

- [x] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [x] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -64,6 +64,12 @@ services:
         - SKIP_TESTS=true
         - SPLUNK_URL=${SPLUNK_URL:-http://splunk:8088}
         - SPLUNK_TOKEN=${SPLUNK_HEC_TOKEN:-token}
+        - REDIS_HOST=redis
+        - REDIS_PORT=6379
+        - REDIS_PASSWORD=password
```
Comment (karahanb) [inline docker-compose.yml:69]: Should we get these from the env variables as well? For example; 
REDIS_PASSWORD=${REDIS_PASSWORD:-password}

**Review by karahanb — APPROVED** [review]
Everything looks great. Thanks Kevin!

File: docker-compose.yml (hunk 1 above)
Comment (pbolduc) [reply docker-compose.yml:69]: I would like to understand the use case of over ridding this in docker-compose.yaml.  The direction I was going is just use sensible defaults for local development only.

Comment (karahanb) [reply docker-compose.yml:69]: I was just thinking about a scenario if the password or host value would be different for dev or test environment.

Comment (pbolduc) [reply docker-compose.yml:69]: We dont use docker-compose when we deploy to OpenShift.

Comment (karahanb) [reply docker-compose.yml:69]: Oh, I see. Makes sense. You may just ignore my comment then.

**Pull request merged**